pkg/synopsysctl/cmd_delete.go

```diff
@@ -137,17 +137,23 @@ var deleteBlackDuckCmd = &cobra.Command{
 		}
 
 		labelSelector := fmt.Sprintf("app=%s, name=%s", util.BlackDuckName, args[0])
-		// delete exposed service
 		svcs, err := util.ListServices(kubeClient, namespace, labelSelector)
 		if err != nil {
 			return fmt.Errorf("couldn't list services in namespace '%s' due to %+v", namespace, err)
 		}
 		for _, svc := range svcs.Items {
+			// delete exposed service
 			if strings.HasSuffix(svc.Name, "-exposed") {
 				if err := util.DeleteService(kubeClient, namespace, svc.Name); err != nil && !k8serrors.IsNotFound(err) {
 					return fmt.Errorf("couldn't delete service '%s' in namespace '%s' due to %+v", svc.Name, namespace, err)
 				}
 			}
+			// delete webui service for versions 2021.8.1 and above, as <name>-blackduck-webui is renamed to <name>-blackduck-ui
+			if strings.HasSuffix(svc.Name, "-blackduck-webui") && (util.CompareVersions(globals.BlackDuckVersion, "2021.8.1") >= 0) {
+				if err := util.DeleteService(kubeClient, namespace, svc.Name); err != nil && !k8serrors.IsNotFound(err) {
```

Comment:
Since it's a delete event, we don't need a version check here.

Reply:
True. No matter what version of Blackduck we are running we would want this service to be deleted